parser/Parser/Parser.py

```diff
@@ -0,0 +1,174 @@
+            line=line.lower()
+        line = re.sub(r"[\u0000-\u0027\u200b]", "-", line)
+        line = re.sub(r"&\w+;", "-", line)
+        line = re.sub(r"[\s!\"#\$%&'()*+,\/:;<=>?@\[\\\]^_`{\|}~¡¢£¤¥¦§¨©ª«¬®¯°±²³´µ¶·¸¹º»¼½¾¿×ˆ˜–—‘’‚“”„†‡•…‰‹›€™\t]", "-", line)
+        line = re.sub(r"-+", "-", line)
+        line = line.strip("-")
+        return line
+
+
+    def add_line(self,line):
+        """to get a normalized string but keeping the language code "lc:" and the "," separators"""
+        if len(line)<=3:return self.normalizing(line)
+        else:
+            lang=line[:2]
+            new_line=lang+":"
+            for word in line[3:].split(","):
+                new_line+=self.normalizing(word,lang)+','
+            new_line=new_line[:-1]
+            return new_line
+
+    def create_headernode(self,header):
+        query="CREATE (node{id:'__header__'" + (",value:'"+ header + "'" if header else "") +"})"
+        self.session.run(query)
+
+    def create_node(self,data):
+        id_query = " CREATE (n{id:'" + data['id'] + "'}) SET n.block_index=" + str(self.block_index) + ", n.block_subindex=0, n.preceding_lines="+str(data['comment'])
+        if data['value'] : entry_query = " SET n.value='" + str(data['value']) + "'"
+        else : entry_query = " SET n.tags=\"" + str(data['tags']) + "\" ,n.tagsid=" + str(data['tagsid']) + " ,n.parents=" + str(data['parent_tag']) + " ,n.properties=" + str(data['properties'])
+        query=id_query+entry_query
+        # print(query)
+        self.session.run(query)
+
+    def reinitialize_data(self):
+        data = { "id":'',
+        "comment":[],
+        "parent_tag":[],
+        "tags":{},
+        "tagsid":[],
+        "properties":[],
+        "value":'',
+        }
+        return data
+
+    def header_harvest(self):
+        h=0
+        header=''
+        for line in self.file_iter():
+            if line :
+                if line[0]=='#':
+                    header+=line
+                else : break
+            h+=1
+        return header,h
+
+    def harvest(self):
+        """Transform data in file to nodes in with neo4j"""
+        index_comm=0
+        index_stopwords=1
+        index_synonyms=1
+        language_code_prefix = re.compile('[a-zA-Z][a-zA-Z]:')
+
+        #header
+        header,next_line = self.header_harvest()
+        yield header
+
+        #the other entries
+        self.block_index = 1 #to count block
+        data = self.reinitialize_data()
+        for line in self.file_iter(next_line):
+
+            if line == '':
+                if data['id']:
+                    yield data
+                    data = self.reinitialize_data()
+                    self.block_index+=1
+            elif line[0]=="#":
+                index_comm+=1
+                data['comment'].append(self.normalizing(line,"en"))
+            elif 'stopword' in line:
+                data['id'] = "stopwords:"+str(index_stopwords)
```

Comment:
I think @alexgarel mentions to raise an exception in Python. I think this will help you understand about exceptions: https://docs.python.org/3/library/exceptions.html